Sources/PriorityQueue/PriorityQueue.swift

```diff
@@ -0,0 +1,227 @@
+        storage[0] = storage.last!
+        storage.removeLast()
+        sinkDown(elementIndex: 0)
+        return removedElement;
+    }
+    
+    /*
+     *  Moves a specified element up the array (away from index zero) until it's in its natural position
+     */
+    private mutating func sinkDown(elementIndex:Int){
+        var currentIndex = elementIndex
+        var leftChild = getLeftChild(index: elementIndex)
+        var rightChild = getRightChild(index: elementIndex)
+        var childToSwap:Int
+        var temp:Element
+        
+        //loops until a swap is not needed or currentIndex has no children
+        while(true){
+            //chooses which child to swap currentIndex with
+            if(leftChild != -1 && rightChild != -1){
+                //both children are present
+                childToSwap = (compare(firstElement: storage[leftChild], secondElement: storage[rightChild]) < 0) ? leftChild : rightChild;
+            }else if(leftChild != -1){
+                //only left child present
+                childToSwap = leftChild
+            }else if(rightChild != -1){
+                //only right child present
+                childToSwap = rightChild
+            }else{
+                //neither children present
+                break;
+            }
+            //swap currentIndex with appropriate child if needed
+            if(compare(firstElement: storage[childToSwap], secondElement: storage[currentIndex]) < 0){
+                temp = storage[currentIndex]
+                storage[currentIndex] = storage[childToSwap]
+                storage[childToSwap] = temp
+                
+                //updates children and current index in case swap caused more problems
+                currentIndex = childToSwap;
+                leftChild = getLeftChild(index: childToSwap)
+                rightChild = getRightChild(index: childToSwap)
+            }else{
+                //swap is not needed
+                break;
+            }
+        }
+        
+        
+    }
+    
+    /*
+     *  Determines the element with the highest/lowest priority, depending on the heap type
+     *  @return  the element with the highest/lowest priority, or nil if no elements exist
+     */
+    public func peek() -> Element?{
+        if(isEmpty() == true){
+            return nil;
+        }
+        return storage[0];
+    }
+    
+    /*
+     *  Determines whether the Priority Queue is full.
+     *  @return  true if the PriorityQueue is full, false otherwise
+     */
+    public func isFull() -> Bool{
+        return false; //The resizable Priority Queue is never full
+    }
+    
+    /*
+     *  Determines whether the Priority Queue is empty.
+     *  @return  true if the Priority Queue is empty, false otherwise
+     */
+    public func isEmpty() -> Bool{
+        return size() == 0;
+    }
+    
+    /*
+     *  Determines the size of the Priority Queue
+     */
+    public func size() -> Int{
```

Comment:
This needs to be a property called `count`.

```suggestion
    public var count: Int {
```